src/openforms/utils/decorators.py

```diff
@@ -65,3 +70,65 @@ def _wrapped_view_func(request, *args, **kwargs):
         return _wrapped_view_func
 
     return decorator
+
+
+def suppress_requests_errors(fields: list[str]):
+    """
+    Add robustness to admin fields making outgoing requests that may fail.
+
+    If a :class:`requests.RequestException`is caught, return the base (non-enhanced)
+    form field that Django would generate by default and append an error message
+    for the end-user.
+
+    :arg fields: A list of model field names for which to suppress errors.
+    """
+
+    def fallback(self, db_field, request, *args, **kwargs):
+        kwargs = {**self.formfield_overrides[db_field.__class__], **kwargs}
+
+        class CustomWidget(kwargs["widget"]):
+            def render(self, *args, **kwargs):
+                output = super().render(*args, **kwargs)
+
+                error_message = _(
+                    "Could not load data - enable and check the request logs for more details"
+                )
+
+                html = format_html(
+                    """
+                    <div class="openforms-error-widget">
+                        <div class="openforms-error-widget__column">
+                            {}
+                            <small class="openforms-error-widget openforms-error-widget--error-text">{}</small>
+                        </div>
+                    </div>""",
+                    output,
+                    error_message,
+                )
+
+                return html
+
+        kwargs["widget"] = CustomWidget
+        return db_field.formfield(**kwargs)
+
+    def _decorator(admin_class: type[admin.ModelAdmin]):
+        original_func = admin_class.formfield_for_dbfield
+        for field in fields:
+            assert admin_class.model._meta.get_field(field)
```

Comment:
I don't like that this requires the `model = ...` to be set on the admin class. I understand why it's done, because moving it inside the `def _wrapped` is too late as a development aid and code coverage cannot help there either.

So I'd propose to pass the `model` as an argument to the `suppress_requests_errors` decorator itself:

```py
@suppress_requests_errors(AppointmentsConfig, fields=["limit_to_location"])
```
which makes performing this sanity check a lot more straightforward, without indirection and polluting the admin class itself.

So, why is this a bad approach? because you can actually register and admin class for multiple models. This is valid:

```py
@admin.register(SOAPService, StUFService, Service)
class MyAdmin(admin.ModelAdmin):
    search_fields = ("label",)
```
i.e. registering the same admin class for different models which have some shared functionality.